crates/core/src/timeline/loader.rs

```diff
@@ -0,0 +1,111 @@
+use log::debug;
+use std::sync::Arc;
+
+#[derive(Debug, Clone)]
+pub enum TimelineLoader {
+    LayoutOneActive(Arc<Storage>),
+    LayoutOneArchived(Arc<Storage>),
+    LayoutTwoActive(Arc<Storage>),
+    LayoutTwoCompacted(Arc<Storage>),
+}
+
+impl TimelineLoader {
+    pub async fn load_instants(
+        &self,
+        selector: &TimelineSelector,
+        desc: bool,
+    ) -> Result<Vec<Instant>> {
+        match self {
+            TimelineLoader::LayoutOneActive(storage) => {
+                let files = storage.list_files(Some(HUDI_METADATA_DIR)).await?;
+                let mut instants = Vec::with_capacity(files.len() / 3);
+
+                for file_info in files {
+                    match selector.try_create_instant(file_info.name.as_str()) {
+                        Ok(instant) => instants.push(instant),
+                        Err(e) => {
+                            debug!(
+                                "Instant not created from file {:?} due to: {:?}",
+                                file_info, e
+                            );
+                        }
+                    }
+                }
+
+                instants.sort_unstable();
+                instants.shrink_to_fit();
+
+                if desc {
+                    Ok(instants.into_iter().rev().collect())
+                } else {
+                    Ok(instants)
+                }
+            }
+            TimelineLoader::LayoutTwoActive(storage) => {
+                let files = storage.list_files(Some(LSM_TIMELINE_DIR)).await?;
+                let mut instants = Vec::new();
+
+                for file_info in files {
+                    if file_info.name.starts_with("history/") || file_info.name.starts_with('_') {
+                        continue;
+                    }
+                    match selector.try_create_instant(file_info.name.as_str()) {
+                        Ok(instant) => instants.push(instant),
+                        Err(e) => {
+                            debug!(
+                                "Instant not created from file {:?} due to: {:?}",
+                                file_info, e
+                            );
+                        }
+                    }
+                }
+                Ok(instants)
+            }
+            _ => Err(CoreError::Unsupported(
+                "Loading from this timeline layout is not implemented yet.".to_string(),
+            )),
+        }
+    }
+
+    pub async fn load_archived_instants(
+        &self,
+        _selector: &TimelineSelector,
+        _desc: bool,
+    ) -> Result<Vec<Instant>> {
+        match self {
+            TimelineLoader::LayoutOneArchived(_) => Err(CoreError::Unsupported(
+                "Archived timeline for layout v1 not implemented yet".to_string(),
+            )),
+            TimelineLoader::LayoutTwoCompacted(_) => Err(CoreError::Unsupported(
+                "Compacted timeline for layout v2 not implemented yet".to_string(),
```

Comment:
totally ok to do archived timeline support separately. reminder to fix the name as it's still archived in V2. also this method may not be pub for hiding the impl.

Reply:
Made archived loading methods non-public and aligned messages to "archived" for v2 as well.